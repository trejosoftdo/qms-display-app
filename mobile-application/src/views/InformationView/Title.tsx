import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { List } from 'react-native-paper';
import { useTranslation } from 'react-i18next';
import { TRANSLATION_DETAILS_KEY } from '../../common/translations/translation-keys';

/**
 * TitleProps defines the props for the Title Component.
 */
interface TitleProps {}

/**
 * A component to show a title
 *
 * @param {TitleProps} props - The props for the Title component.
 */
const Title: React.FC<TitleProps> = (props: TitleProps) => {
  const { t } = useTranslation();
  return (
    <View style={styles.title}>
      <List.Icon {...props} icon="information" />
      <Text style={styles.titleText}>{t(TRANSLATION_DETAILS_KEY)}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  title: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    width: '50%',
  },
  titleText: {
    paddingLeft: 10,
    fontSize: 16
  },
});

export default Title;