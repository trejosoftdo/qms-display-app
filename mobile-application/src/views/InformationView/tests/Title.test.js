import React from 'react';
import Title from '../Title';

describe('Title component', () => {
  it('renders the component as expected', () => {
    expect(renderShallow(<Title />)).toMatchSnapshot();
  });
});
