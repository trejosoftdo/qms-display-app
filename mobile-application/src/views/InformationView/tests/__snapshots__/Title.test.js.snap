// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component renders the component as expected 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "width": "50%",
    }
  }
>
  <List.Icon
    icon="information"
  />
  <Text
    style={
      {
        "fontSize": 16,
        "paddingLeft": 10,
      }
    }
  >
    Translated[translation:details]({})
  </Text>
</View>
`;
